selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/hotupdate/RecreateUpdate.java

```diff
@@ -0,0 +1,171 @@
+import static org.testng.Assert.*;
+
+import com.google.inject.Inject;
+import java.io.IOException;
+import org.eclipse.che.commons.json.JsonHelper;
+import org.eclipse.che.selenium.core.SeleniumWebDriver;
+import org.eclipse.che.selenium.core.TestGroup;
+import org.eclipse.che.selenium.core.executor.OpenShiftCliCommandExecutor;
+import org.eclipse.che.selenium.core.provider.CheTestApiEndpointUrlProvider;
+import org.eclipse.che.selenium.core.requestfactory.CheTestAdminHttpJsonRequestFactory;
+import org.eclipse.che.selenium.core.utils.WaitUtils;
+import org.eclipse.che.selenium.core.utils.process.ProcessAgent;
+import org.eclipse.che.selenium.core.workspace.TestWorkspace;
+import org.eclipse.che.selenium.pageobject.CheTerminal;
+import org.eclipse.che.selenium.pageobject.Ide;
+import org.eclipse.che.selenium.pageobject.Menu;
+import org.eclipse.che.selenium.pageobject.ProjectExplorer;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+@Test(groups = {TestGroup.OSIO, TestGroup.MULTIUSER})
+public class RecreateUpdate {
+  private static final String COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE =
+      "get dc | grep che | awk '{print $2}'";
+
+  @Inject CheTestAdminHttpJsonRequestFactory testUserHttpJsonRequestFactory;
+  @Inject CheTestApiEndpointUrlProvider cheTestApiEndpointUrlProvider;
+  @Inject ProjectExplorer projectExplorer;
+  @Inject OpenShiftCliCommandExecutor openShiftCliCommandExecutor;
+  @Inject private ProcessAgent processAgent;
+  @Inject private TestWorkspace workspace;
+  @Inject private Ide ide;
+  @Inject private SeleniumWebDriver seleniumWebDriver;
+  @Inject private CheTerminal terminal;
+  @Inject private Menu menu;
+
+  private int cheDeploymentBeforeRollout;
+
+  private enum WsMasterStauses {
+    RUNNING,
+    READY_TO_SHUTDOWN,
+    PREPARING_TO_SHUTDOWN,
+    SUSPENDED;
+  }
+
+  @BeforeClass
+  public void setUp() throws IOException {
+
+    cheDeploymentBeforeRollout =
+        parseInt(openShiftCliCommandExecutor.execute(COMMAND_FOR_GETTING_CURRENT_DEPLOYMENT_CHE));
+  }
+
+  @Test
+  public void checkRecreateUpdateStrategy() throws Exception {
+    String ocClientRolloutCommand = "rollout latest che";
+    String restUrlForSuspendingWorkspaces =
+        cheTestApiEndpointUrlProvider.get().toString() + "system/stop";
+    int timeLimitInSecForRecreateingUpdate = 600;
+    int delayBetweenRequest = 6;
+
+    // open a user workspace and send request for preparing to shutdown
+    ide.open(workspace);
+
+    testUserHttpJsonRequestFactory
+        .fromUrl(restUrlForSuspendingWorkspaces)
+        .usePostMethod()
+        .request();
+
+    // make sure, that system is prepared to  shutdown and than ready to shutdown
+    checkExpectedStatusesAndWorkspaceAfterShutDowning();
+
+    // performs rollout
+    openShiftCliCommandExecutor.execute(ocClientRolloutCommand);
+    waitOpenShiftStatus(
+        timeLimitInSecForRecreateingUpdate, delayBetweenRequest, WsMasterStauses.RUNNING);
```

Comment:
In all occurrences of this method changes only "WsMasterStauses" argument, so you can rid of extra arguments and left only "WsMasterStauses".